docs/docs/migration-guide.mdx

```diff
@@ -10,6 +10,37 @@ description: |
 This page contains information about changes between major versions and
 how you can migrate from one version to another.
 
+## Rasa 2.3.3 to Rasa 2.3.4
+
+:::caution
+This is a release **breaking backwards compatibility of machine learning models**.
+It is not possible to load previously trained models if they were trained with `model_confidence=cosine` or
+`model_confidence=inner` setting. Please make sure to retrain the assistant before trying to use it with this improved version.
+
+:::
+
+### Machine Learning Components
+
+Rasa Open Source `2.3.0` introduced the option of using cosine similarities for model confidences by setting
+`model_confidence=cosine`. Some post-release experiments revealed that doing so can change the order of predicted
+labels. Since this is wrong, the option `model_confidence=cosine` has been removed and is no longer available.
```

Comment:
```suggestion
Rasa Open Source `2.3.0` introduced the option of using cosine similarities for model confidences by setting
`model_confidence=cosine`. Some post-release experiments revealed that using `model_confidence=cosine` is wrong as it can change the order of predicted labels. That's why this option was removed in Rasa Open Source version `2.3.4`.
```